### PR TITLE
Runtime changes common

### PR DESCRIPTION
**Pull request by Eitu33**

### CODE REVIEW

File: massa-grpc/src/api.rs
Comment (damip) [inline massa-grpc/src/api.rs:279]: what does this do ?

Comment (Eitu33) [reply massa-grpc/src/api.rs:279]: Coming from the proto doc: "If the chain is in downtime, we will return the last snapshot before the downtime"

This is an API type update it has nothing to do with this PR, I just put a filler so we can use our changes.

Comment (damip) [reply massa-grpc/src/api.rs:279]: ok thanks :)

File: massa-execution-exports/Cargo.toml
Comment (Leo-Besancon) [inline massa-execution-exports/Cargo.toml:30]: /!\ Do not merge until we:
- Merge the massa-sc-runtime common branch into main
- Update the Cargo.toml with a rev of the massa-sc-runtime commit

File: massa-execution-worker/src/interface_impl.rs
Comment (damip) [inline massa-execution-worker/src/interface_impl.rs:630]: here we can use expect to give an error message

File: massa-execution-worker/src/interface_impl.rs
Comment (damip) [inline massa-execution-worker/src/interface_impl.rs:638]: the unwrap here can be triggered by an attacker sending an invalid signature

File: massa-execution-worker/src/interface_impl.rs
Comment (Leo-Besancon) [inline massa-execution-worker/src/interface_impl.rs:706]: Note: to update once this is merged: https://github.com/massalabs/massa/pull/4152

File: massa-execution-worker/src/interface_impl.rs
Comment (Eitu33) [reply massa-execution-worker/src/interface_impl.rs]: fixed

File: massa-execution-worker/src/interface_impl.rs
Comment (Eitu33) [reply massa-execution-worker/src/interface_impl.rs]: fixed

File: massa-execution-worker/src/interface_impl.rs
Comment (Leo-Besancon) [reply massa-execution-worker/src/interface_impl.rs]: Done

**Review by Leo-Besancon — APPROVED** [review]

**Pull request merged**
